src/share/index.ts

```diff
@@ -118,49 +122,62 @@ export async function share(opts?: ShareOptions): Promise<boolean> {
     return shareSent;
 }
 
-async function doShare (type: ShareType, text: string): Promise<boolean> {
-    const urlPattern = /\bhttps?:\/\/[^\s]+/;
-    const urlMatch = text.match(urlPattern);
-    const textNoUrls = text.replace(urlPattern, "").trim();
+const Adapters: { [target: string]: ShareAdapter } = {
+    [ShareType.Facebook]: (text) => {
+        const urlPattern = /\bhttps?:\/\/[^\s]+/;
+        const urlMatch = text.match(urlPattern);
+        const textNoUrls = text.replace(urlPattern, "").trim();
 
-    switch (type) {
-    case ShareType.Facebook:
         openNewTab("https://www.facebook.com/sharer/sharer.php", {
             quote: textNoUrls,
             u: urlMatch ? urlMatch[0] : createLink(),
         });
-        return true;
-
-    case ShareType.Twitter:
+    },
+    [ShareType.Twitter]: (text) => {
```

Comment:
why dont we need to massage the text for twitter and whatsapp? 

can we add some comments to explain why we are doing what we are doing with the text? doesnt need to be too detailed for now. as we go we should add some things in the docs to explain how the platforms are reacting to the share api calls.

also general/minor comment can probably reuse this code..
`        const urlPattern = /\bhttps?:\/\/[^\s]+/;
        const urlMatch = text.match(urlPattern);
        const textNoUrls = text.replace(urlPattern, "").trim();`

Reply:
it's because those two apps only handle posts/messages that consist of text.  Other platforms split their posts to have media embeds/links be contained in separate fields, so there's no need to include the url in the body.